src/Filesystem/Provider/FileStreamAccess.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace ILIAS\Filesystem\Provider;
+
+use ILIAS\Filesystem\Exception\FileAlreadyExistsException;
+use ILIAS\Filesystem\Exception\FileNotFoundException;
+use ILIAS\Filesystem\Exception\IOException;
+use Psr\Http\Message\StreamInterface;
```

Comment:
I think it would be a good idea to create a own stream interface. My idea was to create the same interface as the PSR-7 with the same behaviour of the streams. The only problem I see is the interaction between the PSR-7 stream and our own. (http service) May I need to provide an adapter for that.

@klees If I understood the RAII pattern correctly, it says that I must allocate the resource while creating an object and free the resource at object destruction. That sounds good to me but I would like to give the consumer of the API the possibility to free the resources ahead of object destruction and clean up the resources if needed.